stock_storage_type/models/stock_location.py

```diff
@@ -55,6 +54,98 @@ class StockLocation(models.Model):
+        ],
+        help="technical field: the pending incoming stock.moves in the location",
+    )
+
+    in_move_line_ids = fields.One2many(
+        "stock.move.line",
+        "location_dest_id",
+        domain=[
+            ("state", "in", ("waiting", "confirmed", "partially_available", "assigned"))
+        ],
+        help="technical field: the pending incoming "
+        "stock.move.lines in the location",
+    )
+    children_in_move_line_ids = fields.Many2many(
+        "stock.move.line",
+        _compute="_compute_children_in_move_line_ids",
+        store=True,
+        help="technical field: the pending incoming stock.move.lines "
+        "in the location or children locations",
+    )
+    location_will_contain_lot_ids = fields.Many2many(
+        "stock.production.lot",
+        store=True,
+        compute="_compute_location_will_contain_lot_ids",
+        help="technical field: list of stock.production.lots in "
+        "the location, either now or in pending operations",
+    )
+    location_will_contain_product_ids = fields.Many2many(
+        "product.product",
+        store=True,
+        compute="_compute_location_will_contain_product_ids",
+        help="technical field: list of products in "
+        "the location, either now or in pending operations",
+    )
+
+    @api.depends("in_move_line_ids", "child_ids.children_in_move_line_ids")
+    def _compute_children_in_move_line_ids(self):
+        for rec in self:
+            rec.children_in_move_line_ids = (
+                rec.mapped("child_ids.children_in_move_line_ids") | rec.in_move_line_ids
+            )
+
+    @api.depends(
+        "quant_ids", "in_move_ids", "in_move_line_ids", "children_in_move_line_ids"
+    )
+    def _compute_location_will_contain_product_ids(self):
+        for rec in self:
+            rec.location_will_contain_product_ids = (
+                rec.mapped("quant_ids.product_id")
+                | rec.mapped("in_move_ids.product_id")
+                | rec.mapped("in_move_line_ids.product_id")
+                | rec.mapped("children_in_move_line_ids.product_id")
+            )
+
+    @api.depends("quant_ids", "in_move_line_ids", "children_in_move_line_ids")
+    def _compute_location_will_contain_lot_ids(self):
+        for rec in self:
+            rec.location_will_contain_lot_ids = (
+                rec.mapped("quant_ids.lot_id")
+                | rec.mapped("in_move_line_ids.lot_id")
+                | rec.mapped("children_in_move_line_ids.lot_id")
+            )
+
+    @api.depends(
+        "quant_ids.quantity",
+        "in_move_ids",
+        "in_move_line_ids",
+        "children_in_move_line_ids",
+    )
+    def _compute_location_is_empty(self):
+        for rec in self:
+            if (
+                sum(rec.quant_ids.mapped("quantity"))
+                or rec.in_move_ids
+                or rec.in_move_line_ids
+                or rec.children_in_move_line_ids
```

Comment:
This already includes the above in_move_line_ids

Reply:
Yes, but I removed the children_in_move_line_ids field in the latest fixups (see my last comments), because:

* children_in_move_line_ids was not recomputed, not sure why but didn't investigate further as:
* I don't see any scenario where we are interested in what moves lines goes into the child locations are we are interested only by the leaf locations